### PR TITLE
Surface Feature Adherence CT

### LINKED ISSUE #217: Surface Feature Adherence Template
## Description & Tasks
Addition of new template to cover Surface feature and relationship, and addition of adherence group to allow generalisation with IfcCovering. additionally see issue #62 addressing IfcCovering Documentutation and new Concept Template

- [x] Author Surface Feature Adherence Template
- [x] Create Adherence group template

### Discussion Points

None ATM

### Links
[Miro Element](https://miro.com/app/board/o9J_l5rJCLk=/?moveToWidget=3074457362577131657&cot=14)
[Miro Task Card](https://miro.com/app/board/o9J_l5rJCLk=/?moveToWidget=3074457362911087556&cot=14)

### Decisions 
TBC

### PR DESCRIPTION
**Pull request by AlexBrad1eyCT**
Fixes #217 

added single template for Surface Feature adherence discussion required around confusing documentation of coverings and its relationship to surface features in issues, but this will suffice for the purposes of Surface Feature relationships

### CODE REVIEW

File: IFC4x3/Templates/Object Composition/Surface Feature Adherence/DocTemplateDefinition.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	<Rules>
+		<DocModelRuleAttribute Name="HasSurfaceFeatures">
+			<Rules>
+				<DocModelRuleEntity Name="IfcRelAdheresToElement">
+					<Rules>
+						<DocModelRuleAttribute Name="RelatedSurfaceFeatures">
+							<Rules>
+								<DocModelRuleEntity Name="IfcSurfaceFeature">
+									<Rules>
+										<DocModelRuleAttribute Name="Name">
```
Comment (pjanck) [inline IFC4x3/Templates/Object Composition/Surface Feature Adherence/DocTemplateDefinition.xml:12]: Doesn't one need `Identification` to be able to specify this as a `RuleID` in the mvdXML?

**Review by pjanck — COMMENTED** [review]
Capitalization in Documentation is a puzzle to me. What is the rule there?

File: IFC4x3/Templates/Object Composition/Surface Feature Adherence/DocTemplateDefinition.xml (hunk 1 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Templates/Object Composition/Surface Feature Adherence/DocTemplateDefinition.xml]: Addressed
@SergejMuhic @jmirtsch I can't seem to find how to add this in the interface??

**Review by pjanck — APPROVED** [review]
lftm

**Comment by AlexBrad1eyCT** [comment]
@larswik review would be appreciated because you generated unit tests for this area.

File: IFC4x3/Templates/Object Composition/Surface Feature Adherence/Documentation.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+An adherence indicates an external part composition relationship between the hosting structure, referred to as the "host", and the adhered components, referred to as the "adhering elements". The concept of adherence is used in various ways, examples are:
+
+* Adherence is used on built elements to adhere surface features which modify the hosting element in the form of markings, surface treatments or defects.
```
Comment (larswik) [inline IFC4x3/Templates/Object Composition/Surface Feature Adherence/Documentation.md:3]:
```suggestion
- Adherence is used on built elements to adhere surface features which modify the hosting element in the form of markings, surface treatments or defects.
```

**Review by larswik — COMMENTED** [review]
I wondered if the "*" should be a dash ("-") to serve as an example after the "examples are:". Otherwise lftm.

**Review by larswik — APPROVED** [review]

**Pull request merged**